test/integration/create_classes_test.go

```diff
@@ -0,0 +1,243 @@
+/*
+Copyright 2019 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package integration
+
+import (
+	"context"
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/crossplaneio/crossplane-runtime/pkg/test/integration"
+	crossplaneapis "github.com/crossplaneio/crossplane/apis"
+	clientcmd "k8s.io/client-go/tools/clientcmd"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/yaml"
+
+	"github.com/crossplaneio/stack-gcp/apis"
+	cachev1beta1 "github.com/crossplaneio/stack-gcp/apis/cache/v1beta1"
+	computev1alpha3 "github.com/crossplaneio/stack-gcp/apis/compute/v1alpha3"
+	containerv1beta1 "github.com/crossplaneio/stack-gcp/apis/container/v1beta1"
+	databasev1beta1 "github.com/crossplaneio/stack-gcp/apis/database/v1beta1"
+	storagev1alpha3 "github.com/crossplaneio/stack-gcp/apis/storage/v1alpha3"
+	"github.com/crossplaneio/stack-gcp/apis/v1alpha3"
+	"github.com/crossplaneio/stack-gcp/pkg/controller"
+)
+
+func TestCreateAllClasses(t *testing.T) {
+	cases := map[string]struct {
+		reason string
+		test   func(c client.Client) error
+	}{
+		"CreateProvider": {
+			reason: "A GCP Provider should be created without error.",
+			test: func(c client.Client) error {
+				dat, err := ioutil.ReadFile("../../examples/gcp-provider.yaml")
```

Comment:
I like that we're testing our examples, but I'm a little worried that it might not be obvious to folks that these examples are tied to the integration tests. Is there something we could do to bring them closer together? How will we know which examples are used by integration tests and which are just examples?

Reply:
Ideally we would always validate *all* examples in our integration tests. Therefore, if someone makes a change to `examples/` then integration tests should also be updated. If we need to enforce this in the PR process that could be an option to make sure we do not get out of sync.